### PR TITLE
home work #12

### PR DESCRIPTION
**Pull request by DeliriumInTheVoid**

### CODE REVIEW

File: lesson_12/headers/sort_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+enum class sorting_direction
+{
+    by_rows,
+    by_column
+};
+
+void bubble_sort(int arr[], const int size);
+
+void sort(int arr[][5], const int rows, const int columns, const sorting_direction direction);
```
Comment (helen-black) [inline lesson_12/headers/sort_utils.hpp:11]: Please, create constant and use it here ```arr[][5]```

File: lesson_12/lesson_12.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+void print_menu()
+{
+    std::cout << "1. bubble sort\n";
+    std::cout << "2. quick sort\n";
+    std::cout << "3. Exit\n\n";
+
+    std::cout << "Enter number:";
+}
+
+void bubble_sort()
```
Comment (helen-black) [inline lesson_12/lesson_12.cpp:59]: it's better to chreate some other name for this function

File: lesson_12/lesson_12.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    bubble_sort(arr, size);
+    std::cout << "sorted array:\n";
+    utl::print_arr(arr, size);
+
+    delete[] arr;
+}
+
+
+void q_sort()
```
Comment (helen-black) [inline lesson_12/lesson_12.cpp:82]: the same

File: lesson_12/sources/sort_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,119 @@
+#include "../headers/sort_utils.hpp"
+
+void quick_sort_in_row(int arr[][5], const int low, const int high, const int row);
+int get_pivot_index_in_row(int arr[][5], const int low, const int high, const int row);
+
+void quick_sort_in_column(int arr[][5], const int low, const int high, const int column);
+int get_pivot_index_in_column(int arr[][5], const int low, const int high, const int column);
```
Comment (helen-black) [inline lesson_12/sources/sort_utils.cpp:7]: use constant size here too

File: lesson_12/sources/sort_utils.cpp [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        bool had_swap{ false };
+        for (int j = 0; j < size - 1 - i; ++j)
+        {
+            if (arr[j] > arr[j + 1])
+            {
+                const int tmp{ arr[j] };
+                arr[j] = arr[j + 1];
+                arr[j + 1] = tmp;
+                had_swap = true;
```
Comment (helen-black) [inline lesson_12/sources/sort_utils.cpp:24]: You can use function swap here too

**Review by helen-black — CHANGES_REQUESTED** [review]

File: lesson_12/headers/sort_utils.hpp (hunk 1 above)
Comment (DeliriumInTheVoid) [reply lesson_12/headers/sort_utils.hpp]: done

File: lesson_12/lesson_12.cpp (hunk 2 above)
Comment (DeliriumInTheVoid) [reply lesson_12/lesson_12.cpp]: done

File: lesson_12/lesson_12.cpp (hunk 3 above)
Comment (DeliriumInTheVoid) [reply lesson_12/lesson_12.cpp]: done

File: lesson_12/sources/sort_utils.cpp (hunk 4 above)
Comment (DeliriumInTheVoid) [reply lesson_12/sources/sort_utils.cpp]: done

File: lesson_12/sources/sort_utils.cpp (hunk 5 above)
Comment (DeliriumInTheVoid) [reply lesson_12/sources/sort_utils.cpp]: done

**Pull request merged**